internal/dag/httpproxy_processor.go

```diff
@@ -948,20 +951,21 @@ func (p *HTTPProxyProcessor) computeRoutes(
 			}
 
 			c := &Cluster{
-				Upstream:              s,
-				LoadBalancerPolicy:    lbPolicy,
-				Weight:                uint32(service.Weight),
-				HTTPHealthCheckPolicy: healthPolicy,
-				UpstreamValidation:    uv,
-				RequestHeadersPolicy:  reqHP,
-				ResponseHeadersPolicy: respHP,
-				CookieRewritePolicies: cookieRP,
-				Protocol:              protocol,
-				SNI:                   determineSNI(r.RequestHeadersPolicy, reqHP, s),
-				DNSLookupFamily:       string(p.DNSLookupFamily),
-				ClientCertificate:     clientCertSecret,
-				TimeoutPolicy:         ctp,
-				SlowStartConfig:       slowStart,
+				Upstream:                 s,
+				LoadBalancerPolicy:       lbPolicy,
+				Weight:                   uint32(service.Weight),
+				HTTPHealthCheckPolicy:    healthPolicy,
+				UpstreamValidation:       uv,
+				RequestHeadersPolicy:     reqHP,
+				ResponseHeadersPolicy:    respHP,
+				CookieRewritePolicies:    cookieRP,
+				Protocol:                 protocol,
+				SNI:                      determineSNI(r.RequestHeadersPolicy, reqHP, s),
+				DNSLookupFamily:          string(p.DNSLookupFamily),
+				ClientCertificate:        clientCertSecret,
+				TimeoutPolicy:            ctp,
+				SlowStartConfig:          slowStart,
+				MaxRequestsPerConnection: p.MaxRequestsPerConnection,
```

Comment:
this isn't exclusive to this change, but we have various cluster-level settings that are supposed to be global that do not get applied to Ingress or Gateway API generated clusters

probably worth a follow up issue so we don't block this but we should make this consistent

cc @skriss

Reply:
xref #5458